mne_bids/dig.py

```diff
@@ -188,23 +220,219 @@ def _coordsystem_json(raw, unit, orient, coordsystem_name, fname,
-                    'MEGCoordinateUnits': unit,  # XXX validate this
-                    'HeadCoilCoordinates': coords,
-                    'HeadCoilCoordinateSystem': orient,
-                    'HeadCoilCoordinateUnits': unit  # XXX validate this
-                    }
+        fid_json = {
+            'MEGCoordinateSystem': coordsystem_name,
+            'MEGCoordinateUnits': unit,  # XXX validate this
+            'HeadCoilCoordinates': coords,
+            'HeadCoilCoordinateSystem': orient,
+            'HeadCoilCoordinateUnits': unit  # XXX validate this
+        }
     elif kind == 'eeg':
-        fid_json = {'EEGCoordinateSystem': coordsystem_name,
-                    'EEGCoordinateUnits': unit,
-                    'AnatomicalLandmarkCoordinates': coords,
-                    'AnatomicalLandmarkCoordinateSystem': coordsystem_name,
-                    'AnatomicalLandmarkCoordinateUnits': unit,
-                    }
+        fid_json = {
+            'EEGCoordinateSystem': coordsystem_name,
+            'EEGCoordinateUnits': unit,
+            'AnatomicalLandmarkCoordinates': coords,
+            'AnatomicalLandmarkCoordinateSystem': coordsystem_name,
+            'AnatomicalLandmarkCoordinateUnits': unit,
+        }
     elif kind == "ieeg":
         fid_json = {
             'iEEGCoordinateSystem': coordsystem_name,  # (Other, Pixels, ACPC)
             'iEEGCoordinateUnits': unit,  # m (MNE), mm, cm , or pixels
         }
 
     _write_json(fname, fid_json, overwrite, verbose)
+
+
+def _write_dig_bids(electrodes_fname, coordsystem_fname, data_path,
+                    raw, kind, overwrite=False, verbose=True):
+    """Write BIDS formatted DigMontage from Raw instance.
+
+    Handles coordinatesystem.json and electrodes.tsv writing
+    from DigMontage.
+
+    Parameters
+    ----------
+    electrodes_fname : str
+        Filename to save the electrodes.tsv to.
+    coordsystem_fname : str
+        Filename to save the coordsystem.json to.
+    data_path : str | | pathlib.Path
+        Path to the data directory
+    raw : instance of Raw
+        The data as MNE-Python Raw object.
+    kind : str
+        Type of the data as in ALLOWED_KINDS.
+    overwrite : bool
+        Whether to overwrite the existing file.
+        Defaults to False.
+    verbose : bool
+        Set verbose output to true or false.
+    """
+    # write electrodes data for iEEG and EEG
+    unit = "m"  # defaults to meters
+
+    params = _parse_bids_filename(electrodes_fname, verbose)
+    subject_id = params['sub']
+    session_id = params['ses']
+    acquisition = params['acq']
+
+    # get coordinate frame from digMontage
+    digpoint = raw.info['dig'][0]
+    if any(digpoint['coord_frame'] != _digpoint['coord_frame']
+           for _digpoint in raw.info['dig']):
+        warn("Not all digpoints have the same coordinate frame. "
+             "Skipping electrodes.tsv writing...")
+    else:
```

Comment:
you can also just put a `return` after the warning and get rid of the `else` nesting right below. "flat is better than nested" (python zen)